components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -73,20 +73,22 @@ public String jwks() {
         String tenantDomain = getTenantDomain();
         String keystorePath = CarbonUtils.getServerConfiguration().getFirstProperty(SECURITY_KEY_STORE_LOCATION);
 
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (isInvalidTenantId(tenantId)) {
```

Comment:
do we need to validate the tenant here since we are retrieving the tenant domain from thread local context?

Reply:
Yes. But if someone calls the jwks endpoint of a non-existing tenant domain still it would set in the thread-local and passed here. isn't it?

Reply:
it seems we are already doing the validation in rewrite valve.. https://github.com/wso2-extensions/identity-carbon-auth-rest/blob/3d37c882f4b96701b8d333c2fc82b42b18da0348/components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java#L111-L112
I think we can remove the redundant validation here. WDYT

Reply:
+1

Reply:
:+1: